lib/src/id_prefix.rs

```diff
@@ -176,10 +176,12 @@ impl IdPrefixContext {
                 .change_index
                 .resolve_prefix_to_key(&*indexes.commit_change_ids, prefix);
             if let PrefixResolution::SingleMatch(change_id) = resolution {
-                // There may be more commits with this change id outside the narrower sets.
-                return PrefixResolution::SingleMatch(repo.resolve_change_id(&change_id).expect(
-                    "Change ids present in narrower search set should be present globally.",
-                ));
+                // There may be more commits with this change id outside the
+                // narrower sets. The change id could also be hidden.
+                return match repo.resolve_change_id(&change_id) {
+                    Some(commit_ids) => PrefixResolution::SingleMatch(commit_ids),
+                    None => PrefixResolution::NoMatch,
```

Comment:
nit: Add a comment here. It's quite surprising that this can happen until you consider hidden commits included in the disambiguation set.

Reply:
Split comment and slightly updated. Does it look good?

Reply:
Yep. Thank you